ui/app/src/model/datasource-api.ts

```diff
@@ -11,40 +11,49 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { ProjectDatasource, DatasourceSelector, GlobalDatasource, Datasource } from '@perses-dev/core';
+import { ProjectDatasource, DatasourceSelector, GlobalDatasource } from '@perses-dev/core';
 import { DatasourceApi } from '@perses-dev/dashboards';
 import LRUCache from 'lru-cache';
 import { fetchDatasourceList } from './datasource-client';
 import { fetchGlobalDatasourceList } from './global-datasource-client';
 
-//TODO: Move to tanstack query for caching
 export class HTTPDatasourceAPI implements DatasourceApi {
-  getDatasource(
-    project: string,
-    selector: DatasourceSelector
-  ): Promise<{ resource: ProjectDatasource; proxyUrl: string } | undefined> {
+  /**
+   * Helper function for getting a proxy URL from separate input parameters.
+   * Give the following output according to the definition or not of the input.
+   * - /proxy/globaldatasources/{name}
+   * - /proxy/projects/{project}/datasources/{name}
+   * - /proxy/projects/{project}/dashboards/{dashboard}/{name}
+   *
+   * NB: despite the fact it's possible, it is useless to give a dashboard without a project as the url will for sure
+   * correspond to nothing.
+   * @param props.name
+   * @param props.dashboard
+   * @param props.project
+   */
+  buildProxyUrl(props: { project?: string; dashboard?: string; name: string }): string {
+    const { project, dashboard, name } = props;
+    let url = `${!project && !dashboard ? 'globaldatasources' : 'datasources'}/${name}`;
+    if (dashboard) {
+      url = `dashboards/${dashboard}/${url}`;
+    }
+    if (project) {
+      url = `projects/${project}/${url}`;
+    }
+    return `/proxy/${url}`;
+  }
```

Comment:
@celian-garcia continuing the conversation here as I needed to add more lines to the suggestion. In that case it can be changed as follows: ( I added the encodeURIComponent as this names might contain special characters )
```suggestion
  buildProxyUrl({ project, dashboard, name } : { project?: string; dashboard?: string; name: string }): string {
    let url = `${!project && !dashboard ? 'globaldatasources' : 'datasources'}/${encodeURIComponent(name)}`;
    if (dashboard) {
      url = `dashboards/${encodeURIComponent(dashboard)}/${url}`;
    }
    if (project) {
      url = `projects/${encodeURIComponent(project)}/${url}`;
    }
    return `/proxy/${url}`;
  }
```

Reply:
LGTM thanks, I will do the change

Reply:
Updated the PR with your suggestion + the buildProxyUrl method is now optional, so if you don't use the proxy feature, you don't need to implement it.